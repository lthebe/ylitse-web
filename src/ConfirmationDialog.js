import React from 'react';
import PropTypes from 'prop-types';
import Dialog, {
    DialogActions, DialogContent, DialogContentText, DialogTitle,
} from 'material-ui/Dialog';
import Button from 'material-ui/Button';

const ConfirmationDialog = ({
    onCloseConfirmation, confirmationOpen, label, onAction, buttonLabel,
}) => (
    <Dialog
        open={confirmationOpen}
        onClose={onCloseConfirmation}
    >
        <DialogTitle>Confirm</DialogTitle>
        <DialogContent>
            <DialogContentText>
                {label}
            </DialogContentText>
        </DialogContent>
        <DialogActions>
            <Button onClick={onCloseConfirmation}>
              Cancel
            </Button>
            <Button
                color="secondary"
                onClick={onAction}
            >
                {buttonLabel}
            </Button>
        </DialogActions>
    </Dialog>
);

ConfirmationDialog.propTypes = {
    onCloseConfirmation: PropTypes.func.isRequired,
    confirmationOpen: PropTypes.bool.isRequired,
    label: PropTypes.string.isRequired,
    onAction: PropTypes.func.isRequired,
    buttonLabel: PropTypes.string.isRequired,
};

export default ConfirmationDialog;
