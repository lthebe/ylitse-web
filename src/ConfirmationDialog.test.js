import React from 'react';
import { shallow } from 'enzyme';
import Button from 'material-ui/Button';
import Dialog, { DialogActions } from 'material-ui/Dialog';

import ConfirmationDialog from './ConfirmationDialog';

describe('ConfirmationDialog', () => {
    const onAction = jest.fn();
    const onCloseConfirmation = jest.fn();
    const props = {
        onCloseConfirmation,
        confirmationOpen: false,
        label: 'ConfirmationDialog',
        onAction,
        buttonLabel: 'Delete',
    };
    test('renders correctly', () => {
        const wrapper = shallow(<ConfirmationDialog {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('cancel and delete button called', () => {
        const wrapper = shallow(<ConfirmationDialog {...props} />);
        const cancelButton = wrapper
            .find(Dialog).find(DialogActions).find(Button).at(0);
        cancelButton.simulate('click');
        expect(onCloseConfirmation).toHaveBeenCalled();
        const deleteButton = wrapper
            .find(Dialog).find(DialogActions).find(Button).at(1);
        deleteButton.simulate('click');
        expect(onAction).toHaveBeenCalled();
    });
});
