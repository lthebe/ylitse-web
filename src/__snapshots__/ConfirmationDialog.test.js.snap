// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDialog renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConfirmationDialog
    buttonLabel="Delete"
    confirmationOpen={false}
    label="ConfirmationDialog"
    onAction={[MockFunction]}
    onCloseConfirmation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(DialogTitle)>
          Confirm
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            ConfirmationDialog
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            onClick={[MockFunction]}
          >
            Cancel
          </WithStyles(Button)>
          <WithStyles(Button)
            color="secondary"
            onClick={[MockFunction]}
          >
            Delete
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "onClose": [MockFunction],
      "open": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Confirm",
        },
        "ref": null,
        "rendered": "Confirm",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(DialogContentText)>
            ConfirmationDialog
          </WithStyles(DialogContentText)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "ConfirmationDialog",
          },
          "ref": null,
          "rendered": "ConfirmationDialog",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              onClick={[MockFunction]}
            >
              Cancel
            </WithStyles(Button)>,
            <WithStyles(Button)
              color="secondary"
              onClick={[MockFunction]}
            >
              Delete
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Delete",
              "color": "secondary",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Delete",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(DialogTitle)>
            Confirm
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              ConfirmationDialog
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              onClick={[MockFunction]}
            >
              Cancel
            </WithStyles(Button)>
            <WithStyles(Button)
              color="secondary"
              onClick={[MockFunction]}
            >
              Delete
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction],
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Confirm",
          },
          "ref": null,
          "rendered": "Confirm",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)>
              ConfirmationDialog
            </WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "ConfirmationDialog",
            },
            "ref": null,
            "rendered": "ConfirmationDialog",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                onClick={[MockFunction]}
              >
                Cancel
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="secondary"
                onClick={[MockFunction]}
              >
                Delete
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Delete",
                "color": "secondary",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Delete",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
